src/test/java/com/mentoring/pageobject/CharacterSelectionPageDev.java

```diff
@@ -109,6 +112,8 @@ public void renameCharacter(String newCharacterName) {
 
     @Step("Clicking on Rename character button.")
     public void clickForRenameCharacterButtonInList() {
+        if (characterButtons.size() == 0)
+            waitForMilliSec(200);
```

Comment:
If we only need to wait until the characterButtons has at least one element, then you can use the waitUntilVisible method in Page.java (just make it public) and wait till the first element of the element list appears. No need for static wait then.